audit/src/EthPlaysV0.sol

```diff
@@ -0,0 +1,480 @@
+contract EthPlaysV0 is Ownable {
+    /* -------------------------------------------------------------------------- */
+    /*                                   STRUCTS                                  */
+    /* -------------------------------------------------------------------------- */
+
+    struct ControlBid {
+        address from;
+        uint256 amount;
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                                   STORAGE                                  */
+    /* -------------------------------------------------------------------------- */
+
+    /// @notice [Contract] The POKE token contract
+    Poke public poke;
+    /// @notice [Contract] The EthPlays registry contract
+    RegistryReceiverV0 public registryReceiver;
+
+    /// @notice [Parameter] Indicates if the game is currently active
+    bool public isActive;
+
+    /// @notice [State] The index of the last executed input
+    uint256 public inputIndex;
+    /// @notice [State] The block timestamp of the previous input
+    uint256 private inputTimestamp;
+
+    /// @notice [Parameter] The fraction of alignment to persist upon decay, out of 1000
+    uint256 public alignmentDecayRate;
+    /// @notice [Parameter] Number of seconds between alignment votes for each account
+    uint256 public alignmentVoteCooldown;
+    /// @notice [Parameter] The current reward (in POKE) for voting for chaos
+    uint256 public chaosVoteReward;
+    /// @notice [State] Timestamp of latest alignment vote by account address
+    mapping(address => uint256) private alignmentVoteTimestamps;
+    /// @notice [State] The current alignment value
+    int256 public alignment;
+
+    /// @notice [Parameter] Number of seconds in the order vote period
+    uint256 public orderDuration;
+    /// @notice [State] Count of order votes for each button index, by input index
+    uint256[8] private orderVotes;
+    /// @notice [State] Most recent inputIndex an account submitted an order vote
+    mapping(address => uint256) private inputIndices;
+
+    /// @notice [State] Timestamp of the most recent chaos input for each account
+    mapping(address => uint256) private chaosInputTimestamps;
+    /// @notice [Parameter] Number of seconds of cooldown between chaos rewards
+    uint256 public chaosInputRewardCooldown;
+
+    /// @notice [Parameter] The current reward (in POKE) for chaos inputs, subject to cooldown
+    uint256 public chaosInputReward;
+    /// @notice [Parameter] The current reward (in POKE) for order input votes
+    uint256 public orderInputReward;
+    /// @notice [Parameter] The current cost (in POKE) to submit a chat message
+    uint256 public chatCost;
+    /// @notice [Parameter] The current cost (in POKE) to buy a rare candy
+    uint256 public rareCandyCost;
+
+    /// @notice [Parameter] The number of seconds that the control auction lasts
+    uint256 public controlAuctionDuration;
```

Comment:
Have you considered what might happen to auctions, etc. if the chain experiences a brief interruption where blocks stop for a few minutes? Sammy's orchids NFT had this issue where his "needs water" counter was time-based, then Polygon had some downtime where ~everyone's orchids died (after months of player care). If this was based on block numbers instead, it would resist issues like this.

Reply:
This is interesting, and I hadn't considered it. Thanks for calling it out!

I'm nearly certain that the auction and the reward timeouts/cooldowns don't break if players are unable to submit txns for any period of time. The "game clock" will just continue as usual. So, once players are able to submit txns again, players would immediately be able to receive voting/input rewards, the auction would land in either the "over, needs to be settled" or "not started, waiting for first bid" states.

This does raise the question of how the website should detect/handle downtime, though...

BTW: I did consider using block numbers for these timers, but I found that it was a worse experience. Also IIRC some networks don't have a consistent/target block time, so these timers would kinda break (e.g. Optimism).